src/GuzzleRequestAdapter.php

```diff
@@ -143,112 +178,152 @@ public function getParseNodeFactory(): ParseNodeFactory
-
-                if ($response !== null) {
-                    return $result;
-                }
-                $this->throwFailedResponse($result, $errorMappings);
-                if ($this->is204NoContentResponse($result)) {
-                    return null;
-                }
-                if ($primitiveType === StreamInterface::class) {
-                    return $result->getBody();
-                }
-                $rootParseNode = $this->getRootParseNode($result);
-                if (is_subclass_of($primitiveType, Enum::class)) {
-                    return $rootParseNode->getEnumValue($primitiveType);
+        $span = $this->startTracingSpan($requestInfo, 'sendPrimitiveAsync');
+        $scope = $span->activate();
+        try {
+            $finalResponse = $this->getHttpResponseMessage($requestInfo)->then(
+                function (ResponseInterface $result) use ($primitiveType, $requestInfo, $errorMappings, &$span) {
+                    $response = $this->tryHandleResponse($requestInfo, $result, $errorMappings);
+
+                    if ($response !== null) {
+                        return $result;
+                    }
+                    $this->throwFailedResponse($result, $errorMappings, $span);
+                    if ($this->is204NoContentResponse($result)) {
+                        return null;
+                    }
+                    if ($primitiveType === StreamInterface::class) {
+                        return $result->getBody();
+                    }
+                    $rootParseNode = $this->getRootParseNode($result, $span);
+                    if (is_subclass_of($primitiveType, Enum::class)) {
+                        return $rootParseNode->getEnumValue($primitiveType);
+                    }
+                    switch ($primitiveType) {
+                        case 'int':
+                        case 'long':
+                            return $rootParseNode->getIntegerValue();
+                        case 'float':
+                            return $rootParseNode->getFloatValue();
+                        case 'bool':
+                            return $rootParseNode->getBooleanValue();
+                        case 'string':
+                            return $rootParseNode->getStringValue();
+                        case DateTime::class:
+                            return $rootParseNode->getDateTimeValue();
+                        case DateInterval::class:
+                            return $rootParseNode->getDateIntervalValue();
+                        case Date::class:
+                            return $rootParseNode->getDateValue();
+                        case Time::class:
+                            return $rootParseNode->getTimeValue();
+                        default:
+                            throw new InvalidArgumentException("Unsupported primitive type $primitiveType");
+                    }
```

Comment:
Set an attribute for the value of the primitive type. The attribute name is "com.microsoft.kiota.response.type"

Reply:
Done